src/platform/windows.c

```diff
@@ -1126,7 +1117,7 @@ static bool windows_setup_file_association(
 	}
 	// [hRootKey\OpenRCT2.ext\DefaultIcon]
 	wchar_t szIconW[MAX_PATH];
-	wsprintfW(szIconW, L"\"%s\",%d", dllPathW, iconIndex);
+	StringCchPrintfW(szIconW, MAX_PATH, L"\"%s\",%d", dllPathW, iconIndex);
```

Comment:
@LRFLEW I am talking about the fact that strncpy does not add a null terminator if it has to truncate the string.